lib/tapioca/runtime/reflection.rb

```diff
@@ -255,6 +255,35 @@ def child_module_for_parent_with_name(parent, name)
       def method_defined_by_forwardable_module?(method)
         method.source_location&.first == Object.const_source_location(:Forwardable)&.first
       end
+
+      sig { params(symbol: String).returns(String) }
+      def namespace_of(symbol)
+        symbol_arr = symbol.split("::")
+        return "" if symbol_arr.length == 1
+
+        T.must(symbol_arr[0...-1]).join("::")
+      end
+
+      sig { params(symbol: String).returns(T::Boolean) }
+      def contains_alias?(symbol)
```

Comment:
@paracycle Pointed out that there's already a [method in tapioca that basically does this](https://github.com/Shopify/tapioca/blob/1e3c9ac5ddec6a49b10486095be0baf11e5832b3/lib/tapioca/gem/pipeline.rb#L438-L448) and I could just adapt that to fit this use case. I'll reopen the PR once that's done!